pkg/logs/logs.go

```diff
@@ -116,29 +89,14 @@ func start(ac *autodiscovery.AutoConfig, serverless bool) (*Agent, error) {
 	}
 
 	// setup and start the logs agent
-	if !serverless {
-		// regular logs agent
-		log.Info("Starting logs-agent...")
-		agent = NewAgent(sources, services, processingRules, endpoints)
-	} else {
-		// serverless logs agent
-		log.Info("Starting a serverless logs-agent...")
-		agent = NewServerless(sources, services, processingRules, endpoints)
-	}
+	log.Info("Starting logs-agent...")
+	agent = NewAgent(sources, services, processingRules, endpoints)
```

Comment:
This is the crux of the change. Instead of relying on a variable, we now rely on build tags to determine how to create the logs agent.